### DISCUSSION TITLE
Inter 4.0 - Using it for text-stroke (outline/hollow)

### DISCUSSION THREAD

**Discussion by Jimmy-Jim-J** [rsms/inter#742] [Q&A] [answered]
Hi. First time caller on this thread and Github. 

Experiencing an issue with the text when I text-stroke it.

It's not meant in a derogatory term, but when I text-stroke the typeset, I call the design element a 'bit dirty'. See screen grab. 

I appreciate this end use, might not have been a need. 
 
 I couldn't find any other Q&As regarding this type of text treatment in the Inter thread. 
 
 Has anybody else experienced it, and apart from photoshopping the artwork, have a solution? 
 
 
![Inter_text-Stroke_closeup_130924](https://github.com/user-attachments/assets/cf70fd74-96be-4a5e-96d0-812e934736d4)
![Inter_text-Stroke_130924](https://github.com/user-attachments/assets/74d96c4c-4dfb-4e03-8413-0a91b1a94642)

**Comment by Jimmy-Jim-J** [comment]
Me again:

Here's a screengrab of the issue in action for web:

![image](https://github.com/user-attachments/assets/82d319a3-b665-4950-9e66-aea329afef28)

**Comment by rsms** [comment, marked as answer]
This is the case with all variable fonts. You either need to unify the shapes by applying the boolean union (using for example skia-pathops) or use the static font files which are pre-processed with boolean union (aka "remove overlaps")

**Reply by rsms** [reply]
If you're using a design tool like Figma, Sketch or Illustrator, then apply boolean union (they all have this feature) before applying stroke outline

**Reply by rsms** [reply]
Here's an example in Figma. Note that because of weird quirks of Figma, you need two copies of the text object in a boolean group for it to work (long-standing bug.) https://www.figma.com/design/ib8iTLtizxchh3FLNHDbgN/Untitled

<img width="958" alt="Screen Shot 2024-09-13 at 15 15 30" src="https://github.com/user-attachments/assets/8bf18308-22b5-404b-8111-752173cc9ed8">

**Reply by Jimmy-Jim-J** [reply]
@rsms Amazing thank you. 
None-designer designing! so haven't a clue what 'boolean union' or 'skia-pathops' are, but with those keywords and Claude AI I'm sure I can work it out. Thank you for reviewing, taking the time to respond and the working example too!

**Reply by yisibl** [reply]
skia-pathops: https://skia.org/docs/dev/present/pathops/
